### PR TITLE
[WFCORE-5065] Add an optionally implemented, ServiceLoader loadable i…

### PR DESCRIPTION
**Pull request by bstansberry**
…nterface to do deployment transformation

https://issues.redhat.com/browse/WFCORE-5065

This has no effect unless a dist provides a module with an impl. The idea being that an EE 9 variant of full WildFly would.

### CODE REVIEW

File: server/src/main/java/org/jboss/as/server/deployment/transformation/DeploymentTransformer.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+    /**
+     * Perform transformation of filesystem deployment content.
+     * @param src path of the deployment content to be transformed
+     * @param target path of either the file to which the transformed content should be written or to the directory
+     *               in which it should be written. If the latter that name of the written file will be the same as
+     *               the name of the {@code src} file.
+     * @throws IOException if a problem occurs reading or writing the content
+     */
+    void transform(Path src, Path target) throws IOException ;
```
Comment (jmesnil) [inline server/src/main/java/org/jboss/as/server/deployment/transformation/DeploymentTransformer.java:52]: Should we mention anything for in-place replacement (like it is used above in `deploymentTransformer.transform(deploymentFile.toPath(), deploymentFile.toPath())`?

**Comment by jmesnil** [comment]
@bstansberry I'm fine with merging the PR as it is. My only question is about in-place replacement of files. Should we add a comment in the java doc to let the implementors know that src and target could be the same and they're the ones that have to take that into consideration.

File: server/src/main/java/org/jboss/as/server/deployment/transformation/DeploymentTransformer.java (hunk 1 above)
Comment (bstansberry) [reply server/src/main/java/org/jboss/as/server/deployment/transformation/DeploymentTransformer.java:54]: @jmesnil I just updated the javadoc to clarify. Thanks!

**Comment by jmesnil** [comment]
@bstansberry thanks

**Pull request merged**